torch/__init__.py

```diff
@@ -136,6 +136,9 @@ def set_default_tensor_type(t):
 
     _C._set_default_tensor_type(Tensor)
 
+def set_default_arg_validation(check):
```

Comment:
I added this in the context of clients which use pytorch e.g. pyro. It would be cumbersome for the users to enable `validate_args` for each distribution that they use in their programs. Providing them a single switch or some debug flag seems more useful in my opinion.

If you think that a global flag might not be the best way to go forward, can you suggest some other way to propagate this choice among distributions?

Reply:
I don't see a need to propagate validation logic at a library level, but if you feel this feature is needed I recommend
1. moving this to torch/distributions/utils.py with a global variable `distributions_validte_args`
2. defining a method `torch.distributions.set_default_validate_args()`

This decouples `torch.distributions` from `torch` to make it easier to factor torch.distributions out as a separate library later in the future (as suggested by Adam).

Reply:
@lazypanda1 ok I'm warming up to the idea of a global `torch.distributions.set_default_...()` method :smile:, especially seeing that it can also be used to control checking of method arguments in #133 .